librsync faiture linking

**Describe the bug**
I found and installed the librsync library. It was installed as x64-windows-static since it can only compile into a static library. Then I wrote find_library and find_path in my cmakelists. CMake has successfully found both the library and the path to include. But when compiling, the linker cannot find symbols from Librsync

**Environment**
- OS: Windows 10
- Compiler: MSVC 19.28.29335.0

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install librsync
2. CmakeList.txt
`find_library(RSYNC_LIB "rsync.lib")
find_path(RSUNC_INCLUDE "librsync.h")
add_library(patcher ${SOURCES} ${HEADERS})
target_link_libraries(patcher PRIVATE ${RSYNC_LIB} )
target_include_directories(patcher PUBLIC ${RSUNC_INCLUDE})`
3. Use any function from librsync
4. See linking error

**Expected behavior**
Success linking with librsync

Hi! @NancyLi1013 
i checked and set VCPKG_TARGET_TRIPLET on x64-windows-static or on x64-windows but it still didn't help. The library is not linked anyway.
My cmake output
```
"C:\Program Files\JetBrains\CLion 2020.2.4\bin\cmake\win\bin\cmake.exe" -DCMAKE_BUILD_TYPE=Debug -DVCPKG_TARGET_TRIPLET=x64-windows-static -DCMAKE_TOOLCHAIN_FILE=C:/vcpkg/scripts/buildsystems/vcpkg.cmake -G "CodeBlocks - NMake Makefiles" C:\Projects\untitled
-- The CXX compiler identification is MSVC 19.28.29335.0
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.28.29333/bin/Hostx86/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/BuildTools/VC/Tools/MSVC/14.28.29333/bin/Hostx86/x64/cl.exe - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
C:/vcpkg/installed/x64-windows-static/debug/lib/rsync.lib
C:/vcpkg/installed/x64-windows-static/include
-- Configuring done
-- Generating done
-- Build files have been written to: C:/Projects/untitled/cmake-build-debug

[Finished]
```
My full CMakeLists.txt

```
cmake_minimum_required(VERSION 3.5)

project(test LANGUAGES CXX)

set(CMAKE_CXX_STANDARD 11)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
find_library(RSYNC_LIB "rsync.lib")
find_path(RSYNC_INCLUDE "librsync.h")
message(${RSYNC_LIB})
message(${RSYNC_INCLUDE})

add_executable(${PROJECT_NAME} main.cpp)
target_link_libraries(${PROJECT_NAME} PUBLIC ${RSYNC_LIB})
target_include_directories(${PROJECT_NAME} PUBLIC ${RSYNC_INCLUDE})
```
My main.cpp
```
#include <librsync.h>
using namespace std;

int main() {
  FILE* file = rs_file_open("test.txt", "rb", false);
  rs_file_close(file);
  return 0;
}
```
My Complile output:
```
"C:\Program Files\JetBrains\CLion 2020.2.4\bin\cmake\win\bin\cmake.exe" --build C:\Projects\untitled\cmake-build-debug --target test
Scanning dependencies of target test
[ 50%] Building CXX object CMakeFiles/test.dir/main.cpp.obj
main.cpp
[100%] Linking CXX executable test.exe
LINK Pass 1: command "C:\PROGRA~2\MICROS~1\2019\BUILDT~1\VC\Tools\MSVC\1428~1.293\bin\Hostx86\x64\link.exe /nologo @CMakeFiles\test.dir\objects1.rsp /out:test.exe /implib:test.lib /pdb:C:\Projects\untitled\cmake-build-debug\test.pdb /version:0.0 /machine:x64 /debug /INCREMENTAL /subsystem:console C:\vcpkg\installed\x64-windows-static\debug\lib\rsync.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:CMakeFiles\test.dir/intermediate.manifest CMakeFiles\test.dir/manifest.res" failed (exit code 1120) with the following output:
main.cpp.obj : error LNK2019: unresolved external symbol __imp_rs_file_open referenced in function main
main.cpp.obj : error LNK2019: unresolved external symbol __imp_rs_file_close referenced in function main
test.exe : fatal error LNK1120: 2 unresolved externals
```

--- a/ports/librsync/CONTROL
+++ b/ports/librsync/CONTROL
@@ -1,5 +1,6 @@
 Source: librsync
 Version: 2020-09-16
+Port-Version: 1
 Description: librsync is a library for calculating and applying network deltas, with an interface designed to ease integration into diverse network applications.
 Homepage: http://librsync.sourcefrog.net/
 Supports: !uwp

--- a/ports/librsync/portfile.cmake
+++ b/ports/librsync/portfile.cmake
@@ -33,6 +33,13 @@ if(EXISTS ${CURRENT_PACKAGES_DIR}/debug/lib/rsync.dll)
     file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/rsync.dll ${CURRENT_PACKAGES_DIR}/debug/bin/rsync.dll)
 endif()
 
+if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
+    vcpkg_replace_string(${CURRENT_PACKAGES_DIR}/include/librsync_export.h
+        "#  ifdef LIBRSYNC_STATIC_DEFINE"
+        "#  if 1 /* LIBRSYNC_STATIC_DEFINE */"
+    )
+endif()
+
 file(INSTALL
     ${SOURCE_PATH}/COPYING
     DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright

